<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1121</width>
    <height>791</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="2" rowspan="4">
     <widget class="QGraphicsView" name="scene_graphicsView"/>
    </item>
    <item row="0" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Алгоритмы построения растровых отрезков</string>
      </property>
      <widget class="QPushButton" name="dda_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>ЦДА</string>
       </property>
      </widget>
      <widget class="QPushButton" name="bresenham_pushButton">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>20</y>
         <width>141</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>Алгоритм Брезенхема</string>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="integer_bresenham_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>251</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>Алгоритм Брезенхема (целочисленный)</string>
       </property>
      </widget>
      <widget class="QPushButton" name="step_bresenham_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>251</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>Брезенхем с устранением ступенчатости</string>
       </property>
      </widget>
      <widget class="QPushButton" name="vu_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>120</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Алгоритм Ву</string>
       </property>
      </widget>
      <widget class="QPushButton" name="library_pushButton">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>120</y>
         <width>141</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Библиотечный</string>
       </property>
      </widget>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Ввод координат точек</string>
      </property>
      <widget class="QLabel" name="x_start_label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>111</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Х начала отрезка</string>
       </property>
      </widget>
      <widget class="QLabel" name="y_start_label">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>20</y>
         <width>131</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Y начала отрезка</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="x_start_lineEdit">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>101</width>
         <height>16</height>
        </rect>
       </property>
       <property name="clearButtonEnabled">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="y_start_lineEdit">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>40</y>
         <width>121</width>
         <height>16</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="x_end_label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>101</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>X конца отрезка</string>
       </property>
      </widget>
      <widget class="QLabel" name="y_end_label">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>60</y>
         <width>121</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Y конца отрезка</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="x_end_lineEdit">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>101</width>
         <height>22</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="y_end_lineEdit">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>80</y>
         <width>121</width>
         <height>22</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="angle_label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>91</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Угол поворота </string>
       </property>
      </widget>
      <widget class="QLineEdit" name="angle_lineEdit">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>110</y>
         <width>121</width>
         <height>22</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="add_segment_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>140</y>
         <width>121</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Построить отрезок</string>
       </property>
      </widget>
      <widget class="QPushButton" name="add_spectrum_pushButton">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>140</y>
         <width>121</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Построить спектр</string>
       </property>
      </widget>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QGroupBox" name="groupBox_5">
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="back_step_pushButton">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>20</y>
         <width>121</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Шаг назад</string>
       </property>
      </widget>
      <widget class="QPushButton" name="clear_pushButton">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>90</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Очистить сцену</string>
       </property>
      </widget>
     </widget>
    </item>
    <item row="2" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="title">
       <string>Замеры</string>
      </property>
      <widget class="QLabel" name="line_length_label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>101</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Число повторов</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="line_length_lineEdit">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>30</y>
         <width>111</width>
         <height>22</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="compare_steps_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>231</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Сравнение ступенчатости</string>
       </property>
      </widget>
      <widget class="QPushButton" name="compare_time_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>231</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Сравнение временных характеристик</string>
       </property>
      </widget>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Выбор цвета</string>
      </property>
      <widget class="QLabel" name="line_color_label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QPushButton" name="line_color_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>111</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>Цвет отрезка</string>
       </property>
      </widget>
      <widget class="QPushButton" name="background_color_pushButton">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>90</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Цвет фона</string>
       </property>
      </widget>
      <widget class="QLabel" name="background_color_label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>120</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1121</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="program_menu">
    <property name="title">
     <string>О программе</string>
    </property>
   </widget>
   <widget class="QMenu" name="author_menu">
    <property name="title">
     <string>Об авторе</string>
    </property>
   </widget>
   <widget class="QMenu" name="exit_menu">
    <property name="title">
     <string>Выход</string>
    </property>
   </widget>
   <addaction name="program_menu"/>
   <addaction name="author_menu"/>
   <addaction name="exit_menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
